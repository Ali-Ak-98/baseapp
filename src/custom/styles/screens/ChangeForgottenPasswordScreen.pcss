.pg-change-forgotten-password-screen {
  display: flex;
  justify-content: center;
  height: calc(100vh - 58px);
  margin-top: 50px;
  background-color: var(--accent);

  &__background {
    background-image: url(custom/assets/images/landing/looksComplex/RocketEngine.svg);
    background-position: right 18%;
    background-repeat: no-repeat;
    background-size: cover;
    content: "";
    display: block;
    height: calc(100% - 60px);
    position: absolute;
    transform: translate3d(0,0,0);
    right: 0%;
    width: 1050px;
    z-index: 1;
    opacity: 0.2;
    overflow: hidden;
  }

  .custom-input {
    .form-control {
      height: 50px !important;
    }
  }

  .pg-change-forgotten-password-screen__container {
    background: transparent;
    margin-top: 120px;
    width: calc(var(--gap) * 70.66);
    z-index: 3;

    .cr-email-form {
      background: var(--email-form-bg);
      height: 292px;
      min-height: 292px;

      &__option {
        background: var(--header);
        border-radius: 3px;
      }

      &__options-group {
        width: 100%;
        height: calc(var(--big-gap) * 0.97);
        border-bottom: 2px solid var(--email-form-border);
      }

      &__label {
        background: var(--email-form-bg);
        position: absolute;
        top: 0px;
      }
      &__input {
        input {
          color: var(--color-white);
          font-size: calc(var(--gap) * 2.33);
          border: 1px solid var(--input-block-border-color);
          border-radius: calc(var(--gap) * 0.66);
        }
      }
      &__header {
        width: calc(var(--big-gap) * 6.3);
        height: calc(var(--gap) * 4.33);
        font-style: normal;
        font-weight: normal;
        line-height: calc(var(--gap) * 2.66);
        font-size: calc(var(--gap) * 2.33);
        color: var(--placeholder-text);
        margin: calc(var(--gap) * 0.83);
      }
      &__cros-icon {
        position: absolute;
        display: inline-block;
        right: calc(var(--gap) * 3.3);
        margin-top: calc(var(--gap) * 0.5);
        height: calc(var(--gap) * 2.5);
        width: calc(var(--gap) * 2.5);
        cursor: pointer;
      }
      &__button-wrapper {
        min-height: calc(var(--gap) * 10);

        .btn-primary {
          font-size: 14px;
          letter-spacing: 0.3px;
          line-height: 22px;
          background: var(--green) !important;
          border: none !important;
          color: var(--white);
          font-weight: bold;
          border-radius: 4px;
        }
      }
    }

    .cr-email-form__option-inner {
      background: transparent;
      width: 100%;
      height: calc(var(--gap) * 9.66);
      margin: 0 auto;
      text-align: center;
      text-transform: uppercase;
      line-height: calc(var(--gap) * 9.66);
      font-style: normal;
      font-size: calc(var(--gap) * 2.33);
      color: var(--bright-1);
    }

    .cr-email-form__group--focused {
      .cr-email-form__label {
        color: var(--sign-in-accent);
      }
      input {
        border: 2px solid var(--sign-in-accent) !important;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pg-change-forgotten-password-screen {
    .pg-change-forgotten-password-screen__container {
      margin: calc(var(--gap) * 2.5) calc(var(--gap) * 1.25) 0;
      margin-top: 120px;
    }
  }
}
