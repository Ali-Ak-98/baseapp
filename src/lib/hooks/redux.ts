import { useSelector, shallowEqual } from 'react-redux';

import { ReduxState } from 'src/redux/interfaces';
import { LocalizationContainer } from 'src/units/localization/interfaces';

export function useReduxSelector<T>(selector: (state: ReduxState) => T): T {
    return useSelector(selector, shallowEqual);
}

export function useLocalization(): LocalizationContainer {
    return useReduxSelector((x) => x.localization.text);
}
