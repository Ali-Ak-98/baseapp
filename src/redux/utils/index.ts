export * from './action';
export * from './saga-handler';
export * from './wrapper';
export * from './monolite';
