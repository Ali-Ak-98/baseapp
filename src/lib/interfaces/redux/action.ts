import { Action } from 'redux';

export interface ActionWith<PayloadType = any> extends Action {
    payload: PayloadType;
}
