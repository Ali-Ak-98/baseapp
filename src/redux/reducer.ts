import { combineReducers } from 'redux';
import { publicReducer, userReducer } from '../modules/app';

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
});