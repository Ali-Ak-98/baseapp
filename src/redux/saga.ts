import { all, fork } from 'redux-saga/effects';
import { SagaIterator } from 'redux-saga';

import { localizationSaga } from 'src/units/localization/redux';
import { keepSaga } from 'src/units/keep/redux';
import { commonSaga } from 'src/units/common/redux';

export function* rootSaga(): SagaIterator {
    yield all([fork(localizationSaga)]);
    yield all([fork(keepSaga)]);
    yield all([fork(commonSaga)]);
}
