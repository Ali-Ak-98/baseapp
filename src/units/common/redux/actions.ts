import { CommonActionTypes } from './types';
import { ToastParams } from '../interfaces';
import { ReduxUtil } from '../../../redux/utils';
import { KeyValuePair, KeyBooleanPair } from 'lib/interfaces';

export const CommonActions = {
    init: () => ReduxUtil.getAction(CommonActionTypes.INIT),
    dialog: <T = any>(key: string, options?: T) =>
        ReduxUtil.getAction<KeyValuePair>(CommonActionTypes.DIALOG, { key, value: options }),

    loading: (key: string, value: boolean) =>
        ReduxUtil.getAction<KeyBooleanPair>(CommonActionTypes.LOADING, { key: key, value }),

    showToast: (data?: ToastParams) => ReduxUtil.getAction(CommonActionTypes.SHOW_TOAST, data),

    setInitialized: (data: boolean) => ReduxUtil.getAction(CommonActionTypes.SET_INITIALIZED, data),
};
