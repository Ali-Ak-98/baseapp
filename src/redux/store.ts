import { applyMiddleware, compose, createStore } from 'redux';
import sagaMiddlewareFactory from 'redux-saga';
import { rootReducer } from './reducer';
import { rootSaga } from './saga';

const sagaMiddleware = sagaMiddlewareFactory();

// tslint:disable-next-line:no-any
const composeEnhancer: typeof compose = (window as any)
    .__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    rootReducer,
    composeEnhancer(
        applyMiddleware(
            sagaMiddleware,
        ),
    ),
);

// then run the saga
sagaMiddleware.run(rootSaga);

export {
    store,
};
