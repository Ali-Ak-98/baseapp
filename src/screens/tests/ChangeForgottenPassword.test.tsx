import { shallow } from 'enzyme';
import * as React from 'react';
import { InjectedIntlProps } from 'react-intl';
import { connect, Provider } from 'react-redux';
import { store } from '../../redux';
import { ChangeForgottenPasswordScreen } from '../ChangeForgottenPasswordScreen';

const ChangeForgottenPasswordTab = connect()(ChangeForgottenPasswordScreen);

const setup = (props: Partial<InjectedIntlProps> = {}) =>
    shallow(
        <Provider store={store}>
            <ChangeForgottenPasswordTab/>
        </Provider>,
    );

describe('ChangeForgottenPasswordScreen test', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toBeDefined();
    });
});
