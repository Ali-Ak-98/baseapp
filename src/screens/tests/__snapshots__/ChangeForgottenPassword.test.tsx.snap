// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeForgottenPasswordScreen test should render 1`] = `<Connect(InjectIntl(withRouter(Connect(ChangeForgottenPasswordComponent)))) />`;
